### PR TITLE
Improve UX for "Close mode" settings in editor preference page

### PR DESCRIPTION
**Pull request by praveen-skp**
Issue: #2297 

This pull request adds a "Close mode" frame in the editor preference to make the "Close editors automatically" and "Number of opened editors before closing" grouping clear.

### CODE REVIEW

**Comment by praveen-skp** [comment]
Close mode

The new UI with "Close mode" grouping

**Comment by laeubi** [comment]
`Close mode` as a group name reads a bit strange here, I would expect then having different modes to select but the configuration do not really offer that.

My suggestion would be to merge both settings into one line instead of grouping them e.g.

`[ ] Close oldest if there are more than _____ editors open`

**Comment by BeckerWdf** [comment]
> `Close mode` as a group name reads a bit strange here

What about "Closing behaviour"?

**Comment by BeckerWdf** [comment]
> My suggestion would be to merge both settings into one line instead of grouping them

This is also a good idea.

**Comment by praveen-skp** [comment]
This is how the new UI looks.
Disabled
Enabled

**Comment by BeckerWdf** [comment]
Can the input field made shorter so that the line doesn't get too long. I think place for around 4 chars would be sufficient.

What about changing the text to: "Close oldest editor if there more then ... editor open"

What does "oldest" mean? 
a) The one that was open the longest time, or 
b) the one where the time of last use is most far in the past?

**Comment by praveen-skp** [comment]
I checked the current behaviour and noticed that it's not always the oldest editors that are being closed.

### Use case 1:
If I have the following editors opened in order:
1, 2, 3, 4, 5, 6 (1 being the oldest, focus is on 6)

- Set editor limit to 3
- Open editor 7
- The current available editors are 7, 2, 3

It deletes the latest 3 editors (6, 5, 4) and replaces the oldest editor.

---

### Use case 2:
If I have the following editors opened in order:
1, 2, 3, 4, 5, 6 (1 being the oldest, focus is on 5)

- Set editor limit to 3
- Open editor 7
- The current available editors are 7, 2, 3

It deletes the latest 3 editors (5, 6, 4) and replaces the oldest editor (1).

---

### Use case 3:
If I have the following editors opened in order:
1, 2, 3, 4, 5, 6 (1 being the oldest, focus is on 2)

- Set editor limit to 3
- Open editor 7
- The current available editors are 7, 3, 4

It deletes the latest 3 editors (2, 6, 5) and replaces the oldest editor (1).

---

### Use case 4:
If I have the following editors opened in order:
1, 2, 3 (1 being the oldest, focus is on 3)

- Set editor limit to 3
- Open editor 4
- The current available editors are 4, 2, 3

It replaces the oldest editor (1).

---

### Use case 5:
If I have the following editors opened in order:
1, 2, 3 (1 being the oldest, focus is on 1)

- Set editor limit to 3
- Open editor 4
- The current available editors are 1, 4, 3

It replaces the oldest editor (2).

**Comment by praveen-skp** [comment]
It looks like its not always the oldest editors that are being closed. 
So shall we change the text to something generic like: 
Close editor(s) if there are more than ... editors open

**Comment by BeckerWdf** [comment]
> Close editor(s) if there are more than ... editors open

yes.

**Comment by BeckerWdf** [comment]
> > Close editor(s) if there are more than ... editors open
> 
> yes. Maybe  without the () simply "Close editors if..."

**Comment by praveen-skp** [comment]
Reduced size

Reduced the size of the input field to show two characters, to match the text limit of 2 characters in the code.

**Comment by BeckerWdf** [comment]
merge commit are not allowed in PR. Can you pls. squash and rebase on top of current master please?

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/messages.properties [hunk 1]
```diff
@@ -493,12 +493,13 @@ OpenPerspectiveDialogAction_tooltip=Open Perspective
...
 PreferencePage_noDescription = (No description available)
-PreferencePageParameterValues_pageLabelSeparator = \ >\ 
+PreferencePageParameterValues_pageLabelSeparator = \ >\
 ThemingEnabled = E&nable theming
 ThemeChangeWarningText = Restart for the theme changes to take full effect
-ThemeChangeWarningTitle = Theme Changed 
+ThemeChangeWarningTitle = Theme Changed
 # --- Workbench -----
 WorkbenchPreference_openMode=Open mode
+WorkbenchPreference_closeMode=Close mode
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/messages.properties]: we don't need this text any more. Pls. remove it here and from the messages class

**Review by BeckerWdf — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/messages.properties (hunk 1 above)
Comment (BeckerWdf) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/messages.properties]: I did remove it

**Review by BeckerWdf — APPROVED** [review]
I plan to merge this once CI-Build is happy

**Pull request merged**